Sources/ClientRuntime/Retries/DefaultRetryErrorInfoProvider.swift

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.TimeInterval
+
+public enum DefaultRetryErrorInfoProvider: RetryErrorInfoProvider {
```

Comment:
This type figures out whether an error should be retried or not.

This is a "one-size-fits-all" default implementation that is not specific to AWS and intended for general Smithy-based services.

There will be an AWS-specific implementation in a future SDK PR.